src/snowflake/snowpark/dataframe.py

```diff
@@ -1089,6 +1091,31 @@ def col(self, col_name: str) -> Column:
         else:
             return Column(self._resolve(col_name))
 
+    def col_regex(self, regex: str, case_sensitive: bool = False) -> Column:
+        """Selects column based on the column name specified as a regex and returns a reference to it
+        Args:
+            regex: regular expression used to match columns
+        Examples:
+            >>> df = session.create_dataframe([[1, 2, 3, 4]]).to_df(["col1", "col2_a", "col2_b", "col3"])
+            >>> df.select(df.col_regex("col2_.*")).collect()
+            [Row(COL2_A=2, COL2_B=3)]
+
+        """
+        if not isinstance(regex, str):
+            raise ValueError(
+                f"regex provided to col_regex() must be string, got {type(regex)} with value of {regex} instead."
+            )
+        expressions = []
+        flag = 0 if case_sensitive else re.IGNORECASE
+        for column in self._output:
+            if re.match(
+                unquote_if_quoted(regex), unquote_if_quoted(column.name), flags=flag
+            ):
```

Comment:
Sure, say if we want to use it in case sensitive mode. And we have a column named `"Col2_A"`, we actually want both `Col2_A` and `"Col2_A"` to be able to match it, so removing double quotes and only compare the real column name is necessary

Reply:
we need to run more tests to ascertain what the behavior should be. For example, what if the regex is `^"Col.*`. This should match `"Col2_A"` but not `Col2_A`